ui/apps/platform/src/Containers/MainPage/Navigation/HorizontalSubnav.tsx

```diff
@@ -0,0 +1,234 @@
+                            );
+                        }
+                        case 'parent': {
+                            const { key, title, children } = subnavDescription;
+                            return (
+                                <Dropdown
+                                    key={key}
+                                    isPlain
+                                    onSelect={onSelect}
+                                    isOpen={openDropdownKey === key}
+                                    onOpenChange={(isOpen: boolean) =>
+                                        setOpenDropdownKey(isOpen ? key : null)
+                                    }
+                                    toggle={(toggleRef: React.Ref<MenuToggleElement>) => (
+                                        <NavItem
+                                            isActive={subnavDescription.children.some(
+                                                (child) =>
+                                                    child.type === 'link' &&
+                                                    isActiveLink(pathname, child)
+                                            )}
+                                            onClick={() => onToggleClick(key)}
+                                        >
+                                            <MenuToggle
+                                                ref={toggleRef}
+                                                isExpanded={openDropdownKey === key}
+                                                variant="plainText"
+                                            >
+                                                {typeof title === 'function'
+                                                    ? title(subnavDescriptions)
+                                                    : title}
+                                            </MenuToggle>
+                                        </NavItem>
+                                    )}
+                                    shouldFocusToggleOnSelect
+                                >
+                                    <DropdownList>
+                                        {children.map((child) => {
+                                            if (child.type === 'separator') {
+                                                return (
+                                                    <NavItemSeparator
+                                                        key={child.key}
+                                                        role="listitem"
+                                                    />
+                                                );
+                                            }
+                                            const { content, path, description } = child;
+                                            return (
+                                                <DropdownItem
+                                                    component={'a'}
```

Comment:
It could become `component={getLink(path)}` with potential counterpart in LinkShim.tsx file:

```tsx
export function getLink(to: string) {
    return function LinkShim({
        children,
        ...rest
    }: AnchorHTMLAttributes<HTMLAnchorElement>): ReactElement {
        return (
            <Link {...rest} to={to}>
                {children}
            </Link>
        );
    };
}
```

Reply:
I had considered this, but not tested it. I was concerned with the fact that `getLink()` or equivalent would return a new function reference each time, which is essentially a different component. I've had issues in the past with odd rendering behavior doing something similar.

I thought of creating a map that memoizes the calls so that every invocation with the same `to` value returns the same component function, but that seemed like a lot of additional complexity.

Reply:
Indeed, a higher order component might undo the benefit of `LinkShim` element to solve that problem.

Reply:
Admitting, but not necessarily recommending, my first step was a `LinkShimKludge` element that used `id` prop of `DropdownItem` to smuggle in the path.